src/metrics.rs

```diff
@@ -50,12 +50,7 @@ async fn get_metrics() -> impl Responder {
     }
 }
 
-pub(crate) async fn init_metrics_server() -> anyhow::Result<(), std::io::Error> {
-    let port: u16 = std::env::var("PORT")
```

Comment:
@roshaans with your implementation, you ignored cli args. Please have a look how the proposed solution works